src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -995,18 +996,57 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
                     TD.gas_constant_air(thermo_params, ᶜtsʲs.:(1)) /
                     TD.cv_m(thermo_params, ᶜtsʲs.:(1))
 
-                # Placeholders for jacobian contributions of updraft massflux to grid-mean
-                # Initialize all derivatives to zero
+                # Jacobian contributions of updraft massflux to grid-mean
+
+                d_updraft_mass_flux_d_h_tot = ᶠbidiagonal_matrix_ct3
+                @. d_updraft_mass_flux_d_h_tot =
+                    DiagonalMatrixRow(ᶠwinterp(ᶜJ, ᶜρ) * (ᶠu³ʲs.:(1) - ᶠu³)) ⋅
+                    ᶠinterp_matrix() ⋅
+                    DiagonalMatrixRow(Y.c.sgsʲs.:(1).ρa / ᶜρʲs.:(1))
+
+
+                # Derivative of total energy tendency with respect to updraft MSE
+                #### grid-mean ρe_tot
+                @. ∂ᶜρe_tot_err_∂ᶜρ +=
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅ d_updraft_mass_flux_d_h_tot ⋅
+                    DiagonalMatrixRow(
+                        (
+                            -(1 + ᶜkappa_m) * ᶜspecific.e_tot -
+                            ᶜkappa_m * ∂e_int_∂q_tot * ᶜspecific.q_tot
+                        ) / ᶜρ,
+                    )
+
+                @. ∂ᶜρe_tot_err_∂ᶜρq_tot +=
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅ d_updraft_mass_flux_d_h_tot ⋅
+                    DiagonalMatrixRow((ᶜkappa_m * ∂e_int_∂q_tot) / ᶜρ)
+
+                @. ∂ᶜρe_tot_err_∂ᶜρe_tot +=
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅ d_updraft_mass_flux_d_h_tot ⋅
+                    DiagonalMatrixRow((1 + ᶜkappa_m) / ᶜρ)
 
-                ## grid-mean ρρe_tot
                 ∂ᶜρe_tot_err_∂ᶜmseʲ =
                     matrix[@name(c.ρe_tot), @name(c.sgsʲs.:(1).mse)]
-                ∂ᶜρe_tot_err_∂ᶜmseʲ .= (zero(eltype(∂ᶜρe_tot_err_∂ᶜmseʲ)),)
+                @. ∂ᶜρe_tot_err_∂ᶜmseʲ =
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅ d_updraft_mass_flux_d_h_tot
```

Comment:
```suggestion
                    -(dtγ * ᶜadvdivᵥ_matrix() ⋅ d_updraft_mass_flux_d_h_tot)
```
(I think this is correct)